ops/ruby/devops-framework/DEVOPS_FRAMEWORK.md

```diff
@@ -0,0 +1,78 @@
+# DevOps Framework
+
+## Purpose
+This is a small but flexible framework for automating DevOps actions and ensuring they are
+taken in a traceable, repeatable, and consistent way. It operates by  communicating to Google Cloud Platform
+directly, independently of the existing build, operations, and deployment pipeline(s).
+
+## Usage
+The entry point to all functionality is `devops.rb`, which provides command line and environment
+loading and dispatches to Task classes (not scripts) that do the real work.
+`./devops.rb  -t <task_name> -e <env_file> --task-option-1 value1`
+
+There is also a dynamically-generated help file from the descriptions given in the options.
+```
+$ ruby ./devops.rb  --help
+Usage: devops [options]
+    -t, --task [TASK]                Task to be in in each environment
+    -e, --envs-file [ENVS_FILE]      Path to environments JSON file.
+```
+
+##  Installation
+We need a newish version of Ruby. I'm running 2.6.5 as of this writing, but haven't gone
+back and tested with older versions. We also currently need the `gcloud` tools and an account with
+sufficient privilege account to create service account tokens.
+
+Additionally, you'll need to install a gem or two. Right now, it's
+
+`gem install google-cloud-monitoring` for the monitoring library
+
+## Design Decisions
+I should justify some of the  directions I've taken here, since they're departures to varying
+degrees from how we've traditionally done things.
+
+### A New Ruby Application
+I wrestled a bit with this actually. I haven't seen anything that Ruby can do that Python can't,
+for example, and in general the latter language has much more documentation, a more active
+community, and many more programmers. Ultimately, it would've  been a big reset to change
+code bases and languages all at once, so I'll continue to write the first  version in Ruby until
+I get ~~an excuse~~ a requirement to port to Python. One such requirement could come in the form
+of other teams we depend  on and wish to integrate tightly with using some other language.
+
+### Task Classes Instead of Scripts
+I  decided to make the tasks full  Ruby classes instead of stand-alone scripts for a couple  of reasons.
+First, it ensures that we can minimize usage of global state (which can get away from you easily
+in Ruby). Second, we can now configure things  like log levels in one place.
+
+### Independence of All Things All of Us
+There are no dependencies on any AoU application or build/release code, or anything in
+the utils repository. This allows us to avoid refactoring the legacy Ruby stack, and ensures
```

Comment:
Was there some kind of consensus that the existing utils stuff is now legacy? Or is this PR the proposal to make this canon?